jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/message/AbstractAsn1DecodeMessageJSON.java

```diff
@@ -14,14 +14,22 @@ public abstract class AbstractAsn1DecodeMessageJSON extends AbstractSubscriberPr
 	private Logger logger = LoggerFactory.getLogger(this.getClass());
 
 	protected StringPublisher codecPublisher;
+    protected String payload_start_flag;
 
 	public AbstractAsn1DecodeMessageJSON() {
 		super();
 	}
 
-	public AbstractAsn1DecodeMessageJSON(StringPublisher codecPublisher) {
+	public AbstractAsn1DecodeMessageJSON(StringPublisher codecPublisher, String payload_start_flag) {
 		super();
 		this.codecPublisher = codecPublisher;
+		this.payload_start_flag = payload_start_flag;
+	}
+
+	// Strips the IEEE 1609.2 security header (if it exists) and returns the payload
+	protected String stripDot2Header(String hexString) {
+		hexString = hexString.toLowerCase();
+		return hexString.substring(hexString.indexOf(payload_start_flag), hexString.length());
```

Comment:
If this _doesn't_ exist we'll get a -1 right? Will that substring from -1 to length cause an issue?

Reply:
I have gone ahead and added a check for this scenario and made it so the decoders don't publish the message to the Asn1Decoder if it does happen to occur. They log an error message too. They don't throw an exception since I don't think it is needed.